catalyst/core/callbacks/optimizer.py

```diff
@@ -222,4 +224,166 @@ def on_batch_end(self, runner: IRunner) -> None:
             self._accumulation_counter = 0
 
 
-__all__ = ["OptimizerCallback"]
+class AMPOptimizerCallback(Callback):
```

Comment:
what do you think about adding something like
`IOptimizercallback` - interface for optimizer callbacks and `OptimizerCallback` & `AMPOptimizerCallback` as instances

or maybe add some parameter to `OptimizerCallback` to choose correct `optimizer_step`?